app/src/services/market_maker_factory.ts

```diff
@@ -40,6 +43,38 @@ class MarketMakerFactoryService {
 
     return marketMakerAddress
   }
+
+  getMarkets = async (provider: any): Promise<{ markets: Market[] }> => {
+    const filter: any = this.contract.filters.FixedProductMarketMakerCreation()
+
+    const logs = await provider.getLogs({
+      fromBlock: 1,
+      toBlock: 'latest',
+      ...filter,
+    })
+
+    if (logs.length === 0) {
+      throw new Error(`No FixedProductMarketMakerCreation events found`)
```

Comment:
Why throw an error? It's perfectly fine for the array to be empty, we just have to handle this in the UI ("There are no markets...").

Reply:
I can change it, I will return an empty array